test/integration/authorization_test.go

```diff
@@ -276,23 +277,37 @@ type subjectAccessReviewTest struct {
 }
 
 func (test subjectAccessReviewTest) run(t *testing.T) {
-	actualResponse, err := test.clientInterface.Create(test.review)
-	if len(test.err) > 0 {
-		if err == nil {
-			t.Errorf("%s: Expected error: %v", test.description, test.err)
-		} else if !strings.HasPrefix(err.Error(), test.err) {
-			t.Errorf("%s: expected\n\t%v\ngot\n\t%v", test.description, test.err, err)
+	failMessage := ""
+	err := wait.Poll(testutil.PolicyCachePollInterval, testutil.PolicyCachePollTimeout, func() (bool, error) {
+		actualResponse, err := test.clientInterface.Create(test.review)
+		if len(test.err) > 0 {
+			if err == nil {
+				return false, fmt.Errorf("%s: Expected error: %v", test.description, test.err)
```

Comment:
> I'm having a hard time seeing how the poll is helping if the first time we encounter an error, we return it, short-circuiting the poll

An error from an SAR doesn't mean the same thing as a disallowed result from an SAR.  Polling prevents errors where you expected the result of an SAR to be allowed, but it's currently denied because the policy rule hasn't taken effect and vice-versa.  It does not change the cases where you were expecting to be denied access to the SAR altogether, those already worked fine.